### PR TITLE
add codacy coverage support

### PR DESCRIPTION
**Pull request by tduehr**
Adds support for Codacy's test coverage.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -46,17 +46,24 @@ if (!Boolean.getBoolean("skipSonarqube")) {
...
 
+configurations {
+    codacy
+}
+
 buildscript {
     repositories {
         mavenLocal()
         jcenter()
+        maven { url "https://jitpack.io" }
```
Comment (mmoayyed) [inline build.gradle]: Any chance this can be removed?

File: build.gradle [hunk 2]
```diff
@@ -46,17 +46,24 @@ if (!Boolean.getBoolean("skipSonarqube")) {
...
+        maven { url "http://dl.bintray.com/typesafe/maven-releases" }
         maven { url "https://maven.eveoh.nl/content/repositories/releases" }
         maven { url "https://plugins.gradle.org/m2/" }
         maven { url "https://repo.spring.io/plugins-release" }
         maven { url "https://repo.spring.io/libs-milestone" }
     }
 
     dependencies {
+        classpath 'com.github.codacy:codacy-coverage-reporter:-SNAPSHOT'
```
Comment (mmoayyed) [inline build.gradle]: Could we instead build against a specific commit id? or entirely remove jitpack and build against a published binary? (The service generally is very flaky).

Comment (mmoayyed) [reply build.gradle]: PS Also please keep consistent with other classpath entries by using `"` instead of `'`

**Comment by tduehr** [comment]
codacy's coverage is not ready for prime time.

**Pull request closed without merging**